kubernetes/kubernetes_service.go

```diff
@@ -32,7 +34,16 @@ func (in *IstioClient) GetNamespace(namespace string) (*core_v1.Namespace, error
 // It returns a list of all namespaces of the cluster.
 // It returns an error on any problem.
 func (in *IstioClient) GetNamespaces() ([]core_v1.Namespace, error) {
-	namespaces, err := in.k8s.CoreV1().Namespaces().List(emptyListOptions)
+        var listOptions meta_v1.ListOptions;
```

Comment:
I'm not it's appropriate to inject Kiali config here.  This layer, I believe, should be constrained to just k8s api interaction.  It may be better to move this logic to kiali/business/namespaces.go#GetNamespaces().  Or, perhaps we could add a selector argument to this API call.

Reply:
Hmm, ok, so we would just move getting the config to the business logic part and then pass the ListOptions as a parameter to this function?

Reply:
Instead of ListOptions I'd probably suggest just adding a labelSelector argument.  Then create the ListOptions under the covers.  And then something similar in GetProjects().  Basically, just have the caller supply it as opposed to accessing the kiali config in this package.